core/src/main/java/org/apache/accumulo/core/util/Retry.java

```diff
@@ -201,14 +202,14 @@ protected void sleep(Duration wait) throws InterruptedException {
 
   public void logRetry(Logger log, String message, Throwable t) {
     // log the first time as debug, and then after every logInterval as a warning
-    long now = System.nanoTime();
+    NanoTime now = NanoTime.now();
     if (hasNeverLogged) {
       if (log.isDebugEnabled()) {
         log.debug(getMessage(message, t));
       }
       hasNeverLogged = false;
       lastRetryLog = now;
-    } else if ((now - lastRetryLog) > logInterval.toNanos()) {
+    } else if (now.subtract(lastRetryLog).compareTo(logInterval) > 0) {
```

Comment:
This line is just as hard to understand before the change as it is after.
I think this is trying to express something like:

```java
    } else if (lastRetryLog.hasElapsed(logInterval)) {
      log.warn(getMessage(message), t);
      lastRetryLog().reset();
      hasLoggedWarn = true;
```